src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -932,6 +932,24 @@ void uct_ib_mlx5_destroy_qp(uct_ib_mlx5_md_t *md, uct_ib_mlx5_qp_t *qp)
     }
 }
 
+size_t uct_ib_mlx5_sq_length(const uct_ib_mlx5_md_t *md, size_t tx_qp_length,
+                             enum ibv_qp_type qp_type)
+{
+    if (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX) {
+        return ucs_roundup_pow2_or0(tx_qp_length * UCT_IB_MLX5_MAX_BB);
+    } else
+#if HAVE_MLX5_DV
+    if (qp_type == IBV_QPT_DRIVER) {
+        return tx_qp_length * md->tx_wqe_ratio.dc;
+    } else if (qp_type == IBV_QPT_UD) {
+        return tx_qp_length * md->tx_wqe_ratio.ud;
+    } else
+#endif
+    {
+        ucs_fatal("unknown type of MD");
```

Comment:
it can also be wrong qp type. Please add `md_flags` and `qp_type` to this print

Reply:
fixed